<!DOCTYPE html>
<html>
<head>
	<meta charset="utf-8" />
	<meta http-equiv="X-UA-Compatible" content="IE=edge"><title>Visualization of Volcano Plots in R - Samuel&#39;s Blog</title><meta name="viewport" content="width=device-width, initial-scale=1">
	<meta property="og:title" content="Visualization of Volcano Plots in R" />
<meta property="og:description" content="A short tutorial about how to make volcano plots in R." />
<meta property="og:type" content="article" />
<meta property="og:url" content="/post/visualizing-volcano-plots-in-r/" />
<meta property="article:published_time" content="2020-08-09T00:00:00+00:00" />
<meta property="article:modified_time" content="2020-08-09T00:00:00+00:00" />
<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="Visualization of Volcano Plots in R"/>
<meta name="twitter:description" content="A short tutorial about how to make volcano plots in R."/>
<link href="https://fonts.googleapis.com/css?family=Ubuntu:300,400,300italic,400italic|Raleway:200,300" rel="stylesheet">

	<link rel="stylesheet" type="text/css" media="screen" href="/css/normalize.css" />
	<link rel="stylesheet" type="text/css" media="screen" href="/css/main.css" />

	<script src="https://cdn.jsdelivr.net/npm/feather-icons/dist/feather.min.js"></script>
	<script src="/js/main.js"></script>
</head>

<body>
	<div class="container wrapper post">
		<div class="header">
	<base href="/">
	<h1 class="site-title"><a href="/">Samuel&#39;s Blog</a></h1>
	<div class="site-description"><h2>Posts about biology, bioinformatics, and data science</h2><nav class="nav social">
			<ul class="flat"><a href="https://github.com/sdgamboa" title="Github"><i data-feather="github"></i></a><a href="https://twitter.com/samueldgamboa" title="Twitter"><i data-feather="twitter"></i></a><a href="/index.xml" title="RSS"><i data-feather="rss"></i></a></ul>
		</nav>
	</div>

	<nav class="nav">
		<ul class="flat">
			
<li>
  <a href="/" >Home  </a>
  </li>
			
<li>
  <a href="/post" >All posts  </a>
  </li>
			
<li>
  <a href="/tags" >Tags  </a>
  </li>
			
<li>
  <a href="/about" >About  </a>
  </li>
			
<li>
  <a href="https://sdgamboa.github.io/" >Academic Website  </a>
  </li>
			
		</ul>
	</nav>
</div>





		<div class="post-header">
			<h1 class="title">Visualization of Volcano Plots in R</h1>
			<span class="meta" style="font-size:17px;">  Samuel David Gamboa-Tuz  &mdash; August 9, 2020 </span>
			<div class="meta"> </div>
		</div>

		<div class="markdown">
			<h2 id="introduction">Introduction</h2>
<p>A <strong>volcano plot</strong> is a type of scatter plot commonly used in biology research to represent changes in the expression of hundreds or thousands of genes between samples. It&rsquo;s the graphical representation of a differental expression analysis, which can be done with tools like EdgeR or DESeq2. Volcano plots indicate the <strong>fold change</strong> (either positive or negative) in the x axis and a <strong>significance value</strong> (such as the p-value or the adjusted p-value, i.e. FDR) in the y axis. Points represent individual genes and can be labeled or colored according to some attribute, such as whether they are up- or down-regulated, a significance threshold, etc.</p>
<p>In this post I&rsquo;ll go through a step-by-step simple tutorial for the visualization of volcano plots in <strong>R</strong> using tools from the <strong>tidyverse</strong>, such as <strong>dplyr</strong>, <strong>tidyr</strong>, and <strong>ggplot2</strong>. I assume the reader already knows the basics of R and has some familiarity with the tidyverse package.</p>
<h2 id="r-packages-and-data">R packages and data</h2>
<p>In addition to the tidyverse package, I&rsquo;ll load the <strong>ggrepel</strong> package to aid in the labeling of the genes. The example data comes from a <a href="https://doi.org/10.1038/s41598-018-32904-2">research article</a> and can be downloaded <a href="https://raw.githubusercontent.com/sdgamboa/misc_datasets/master/L0_vs_L20.tsv">here</a>.</p>
<p>Let&rsquo;s load the packages and import the data:</p>
<div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r"><span style="color:#60a0b0;font-style:italic"># The packages can be installed within R with:</span>
<span style="color:#60a0b0;font-style:italic"># intstall.packages(c(&#34;tidyverse&#34;, &#34;ggrepel&#34;))</span>

<span style="color:#60a0b0;font-style:italic"># Load packages</span>
<span style="color:#06287e">library</span>(tidyverse)
<span style="color:#06287e">library</span>(ggrepel)

<span style="color:#60a0b0;font-style:italic"># Import data</span>
data <span style="color:#666">&lt;-</span> <span style="color:#06287e">read_tsv</span>(<span style="color:#4070a0">&#34;https://raw.githubusercontent.com/sdgamboa/misc_datasets/master/L0_vs_L20.tsv&#34;</span>)
data
</code></pre></div><div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">## # A tibble: 20,945 x 4
##    Genes   logFC   PValue      FDR
##    &lt;chr&gt;   &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
##  1 276_4  -10.5  1.60e-61 3.35e-57
##  2 62_157  -9.32 6.98e-54 7.31e-50
##  3 144_7   12.2  2.42e-53 1.69e-49
##  4 198_6  -10.0  3.31e-53 1.73e-49
##  5 829_3   11.9  2.98e-51 1.25e-47
##  6 44_124  -8.91 2.98e-50 1.04e-46
##  7 25_31   -9.00 6.55e-50 1.96e-46
##  8 12_195  -8.94 7.51e-50 1.97e-46
##  9 127_41   7.95 1.45e-49 3.38e-46
## 10 31_39    7.72 1.42e-47 2.97e-44
## # â€¦ with 20,935 more rows
</code></pre></div><p>The data contains four columns and 20,945 rows. Each row represents a gene. The column &lsquo;Genes&rsquo; contains the sequence names of each gene. The &lsquo;logFC&rsquo; column contains the logarithm base 2 (log2) of the fold change of each gene; up-regulated genes are positive, down-regulated genes are negative. The &lsquo;PValue&rsquo; and &lsquo;FDR&rsquo; columns contain the significance values; these must be converted to the negative of their logarithm base 10 before plotting, i.e -log10(p-value) or -log10(FDR). I&rsquo;ll use the -log10(FDR) in this post.</p>
<h2 id="a-simple-volcano-plot">A simple volcano plot</h2>
<p>Since volacno plots are scatter plots, we can use <code>geom_point()</code> to generate one with ggplot2. The conversion of the FDR values to their -log10 can be done at this step:</p>
<div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r">p1 <span style="color:#666">&lt;-</span> <span style="color:#06287e">ggplot</span>(data, <span style="color:#06287e">aes</span>(logFC, <span style="color:#666">-</span><span style="color:#06287e">log</span>(FDR,<span style="color:#40a070">10</span>))) <span style="color:#666">+</span> <span style="color:#60a0b0;font-style:italic"># -log10 conversion  </span>
  <span style="color:#06287e">geom_point</span>(size <span style="color:#666">=</span> <span style="color:#40a070">2</span><span style="color:#666">/</span><span style="color:#40a070">5</span>) <span style="color:#666">+</span>
  <span style="color:#06287e">xlab</span>(<span style="color:#06287e">expression</span>(<span style="color:#4070a0">&#34;log&#34;</span>[2]<span style="color:#666">*</span><span style="color:#4070a0">&#34;FC&#34;</span>)) <span style="color:#666">+</span> 
  <span style="color:#06287e">ylab</span>(<span style="color:#06287e">expression</span>(<span style="color:#4070a0">&#34;-log&#34;</span>[10]<span style="color:#666">*</span><span style="color:#4070a0">&#34;FDR&#34;</span>))
p1
</code></pre></div><p><img src="/post/2020-08-09-visualizing-volcano-plots-in-r_files/figure-html/plot 1-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>The dispersion of the points (representing genes) in the plot is similar to, you guessed, a volcano (that&rsquo;s why they&rsquo;re called volcano plots). Since the FDR values were transformed to their -log10, the higher the position of a point, the more significant its value is (y axis). Points with positive fold change values (to the right) are up-regulated and points with negative fold change values (to the left) are down-regulated (x axis).</p>
<h2 id="adding-color-to-differentially-expressed-genes-degs">Adding color to differentially expressed genes (DEGs)</h2>
<p>Differentially expressed genes (DEGs) are usally considered as those with an absolute fold change greater or equal to 2 and a FDR value of 0.05 or less. So, we can make our volcano plot a bit more informative if we add some color to the DEGs in the plot. To do so, we&rsquo;ll add an additional column, named &lsquo;Expression&rsquo;, indicating whether the expression of a gene is up-regulated, down-regulated, or unchanged:</p>
<div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r">data <span style="color:#666">&lt;-</span> data <span style="color:#666">%&gt;%</span> 
  <span style="color:#06287e">mutate</span>(
    Expression <span style="color:#666">=</span> <span style="color:#06287e">case_when</span>(logFC <span style="color:#666">&gt;=</span> <span style="color:#06287e">log</span>(<span style="color:#40a070">2</span>) <span style="color:#666">&amp;</span> FDR <span style="color:#666">&lt;=</span> <span style="color:#40a070">0.05</span> <span style="color:#666">~</span> <span style="color:#4070a0">&#34;Up-regulated&#34;</span>,
                           logFC <span style="color:#666">&lt;=</span> <span style="color:#666">-</span><span style="color:#06287e">log</span>(<span style="color:#40a070">2</span>) <span style="color:#666">&amp;</span> FDR <span style="color:#666">&lt;=</span> <span style="color:#40a070">0.05</span> <span style="color:#666">~</span> <span style="color:#4070a0">&#34;Down-regulated&#34;</span>,
                           <span style="color:#007020;font-weight:bold">TRUE</span> <span style="color:#666">~</span> <span style="color:#4070a0">&#34;Unchanged&#34;</span>)
    )
data
</code></pre></div><div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">## # A tibble: 20,945 x 5
##    Genes   logFC   PValue      FDR Expression    
##    &lt;chr&gt;   &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;         
##  1 276_4  -10.5  1.60e-61 3.35e-57 Down-regulated
##  2 62_157  -9.32 6.98e-54 7.31e-50 Down-regulated
##  3 144_7   12.2  2.42e-53 1.69e-49 Up-regulated  
##  4 198_6  -10.0  3.31e-53 1.73e-49 Down-regulated
##  5 829_3   11.9  2.98e-51 1.25e-47 Up-regulated  
##  6 44_124  -8.91 2.98e-50 1.04e-46 Down-regulated
##  7 25_31   -9.00 6.55e-50 1.96e-46 Down-regulated
##  8 12_195  -8.94 7.51e-50 1.97e-46 Down-regulated
##  9 127_41   7.95 1.45e-49 3.38e-46 Up-regulated  
## 10 31_39    7.72 1.42e-47 2.97e-44 Up-regulated  
## # â€¦ with 20,935 more rows
</code></pre></div><p>We can now map the column Â´Expression&rsquo; to the color aesthetic of <code>geom_point()</code>, and color the points according to their expression classification:</p>
<div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r">p2 <span style="color:#666">&lt;-</span> <span style="color:#06287e">ggplot</span>(data, <span style="color:#06287e">aes</span>(logFC, <span style="color:#666">-</span><span style="color:#06287e">log</span>(FDR,<span style="color:#40a070">10</span>))) <span style="color:#666">+</span>
  <span style="color:#06287e">geom_point</span>(<span style="color:#06287e">aes</span>(color <span style="color:#666">=</span> Expression), size <span style="color:#666">=</span> <span style="color:#40a070">2</span><span style="color:#666">/</span><span style="color:#40a070">5</span>) <span style="color:#666">+</span>
  <span style="color:#06287e">xlab</span>(<span style="color:#06287e">expression</span>(<span style="color:#4070a0">&#34;log&#34;</span>[2]<span style="color:#666">*</span><span style="color:#4070a0">&#34;FC&#34;</span>)) <span style="color:#666">+</span> 
  <span style="color:#06287e">ylab</span>(<span style="color:#06287e">expression</span>(<span style="color:#4070a0">&#34;-log&#34;</span>[10]<span style="color:#666">*</span><span style="color:#4070a0">&#34;FDR&#34;</span>)) <span style="color:#666">+</span>
  <span style="color:#06287e">scale_color_manual</span>(values <span style="color:#666">=</span> <span style="color:#06287e">c</span>(<span style="color:#4070a0">&#34;dodgerblue3&#34;</span>, <span style="color:#4070a0">&#34;gray50&#34;</span>, <span style="color:#4070a0">&#34;firebrick3&#34;</span>)) <span style="color:#666">+</span>
  <span style="color:#06287e">guides</span>(colour <span style="color:#666">=</span> <span style="color:#06287e">guide_legend</span>(override.aes <span style="color:#666">=</span> <span style="color:#06287e">list</span>(size<span style="color:#666">=</span><span style="color:#40a070">1.5</span>))) 
p2
</code></pre></div><p><img src="/post/2020-08-09-visualizing-volcano-plots-in-r_files/figure-html/unnamed-chunk-2-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>If we want to know how many genes are up- or down-regulated, or unchanged, we can use dplyr&rsquo;s <code>count()</code> function.</p>
<div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r">data <span style="color:#666">%&gt;%</span> 
  <span style="color:#06287e">count</span>(Expression)
</code></pre></div><div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">## # A tibble: 3 x 2
##   Expression         n
##   &lt;chr&gt;          &lt;int&gt;
## 1 Down-regulated  2559
## 2 Unchanged      16317
## 3 Up-regulated    2069
</code></pre></div><p>Since we already know that the genes towards the right are up-regulated and the genes towards the left are down-regulated, it would be more informative if we colored the points according to their significance level instead. Let&rsquo;s create another column, named &lsquo;Significance&rsquo;, and classify the genes according to significance thresholds (0.05, 0.01, and 0.001):</p>
<div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r">data <span style="color:#666">&lt;-</span> data <span style="color:#666">%&gt;%</span> 
  <span style="color:#06287e">mutate</span>(
    Significance <span style="color:#666">=</span> <span style="color:#06287e">case_when</span>(
      <span style="color:#06287e">abs</span>(logFC) <span style="color:#666">&gt;=</span> <span style="color:#06287e">log</span>(<span style="color:#40a070">2</span>) <span style="color:#666">&amp;</span> FDR <span style="color:#666">&lt;=</span> <span style="color:#40a070">0.05</span> <span style="color:#666">&amp;</span> FDR <span style="color:#666">&gt;</span> <span style="color:#40a070">0.01</span> <span style="color:#666">~</span> <span style="color:#4070a0">&#34;FDR 0.05&#34;</span>, 
      <span style="color:#06287e">abs</span>(logFC) <span style="color:#666">&gt;=</span> <span style="color:#06287e">log</span>(<span style="color:#40a070">2</span>) <span style="color:#666">&amp;</span> FDR <span style="color:#666">&lt;=</span> <span style="color:#40a070">0.01</span> <span style="color:#666">&amp;</span> FDR <span style="color:#666">&gt;</span> <span style="color:#40a070">0.001</span> <span style="color:#666">~</span> <span style="color:#4070a0">&#34;FDR 0.01&#34;</span>,
      <span style="color:#06287e">abs</span>(logFC) <span style="color:#666">&gt;=</span> <span style="color:#06287e">log</span>(<span style="color:#40a070">2</span>) <span style="color:#666">&amp;</span> FDR <span style="color:#666">&lt;=</span> <span style="color:#40a070">0.001</span> <span style="color:#666">~</span> <span style="color:#4070a0">&#34;FDR 0.001&#34;</span>, 
      <span style="color:#007020;font-weight:bold">TRUE</span> <span style="color:#666">~</span> <span style="color:#4070a0">&#34;Unchanged&#34;</span>)
  )
data
</code></pre></div><div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">## # A tibble: 20,945 x 6
##    Genes   logFC   PValue      FDR Expression     Significance
##    &lt;chr&gt;   &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;          &lt;chr&gt;       
##  1 276_4  -10.5  1.60e-61 3.35e-57 Down-regulated FDR 0.001   
##  2 62_157  -9.32 6.98e-54 7.31e-50 Down-regulated FDR 0.001   
##  3 144_7   12.2  2.42e-53 1.69e-49 Up-regulated   FDR 0.001   
##  4 198_6  -10.0  3.31e-53 1.73e-49 Down-regulated FDR 0.001   
##  5 829_3   11.9  2.98e-51 1.25e-47 Up-regulated   FDR 0.001   
##  6 44_124  -8.91 2.98e-50 1.04e-46 Down-regulated FDR 0.001   
##  7 25_31   -9.00 6.55e-50 1.96e-46 Down-regulated FDR 0.001   
##  8 12_195  -8.94 7.51e-50 1.97e-46 Down-regulated FDR 0.001   
##  9 127_41   7.95 1.45e-49 3.38e-46 Up-regulated   FDR 0.001   
## 10 31_39    7.72 1.42e-47 2.97e-44 Up-regulated   FDR 0.001   
## # â€¦ with 20,935 more rows
</code></pre></div><p>Again, we can use the color aesthetic to map the color of the points to their corresponding significance thresholds:</p>
<div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r">p3 <span style="color:#666">&lt;-</span> <span style="color:#06287e">ggplot</span>(data, <span style="color:#06287e">aes</span>(logFC, <span style="color:#666">-</span><span style="color:#06287e">log</span>(FDR,<span style="color:#40a070">10</span>))) <span style="color:#666">+</span>
  <span style="color:#06287e">geom_point</span>(<span style="color:#06287e">aes</span>(color <span style="color:#666">=</span> Significance), size <span style="color:#666">=</span> <span style="color:#40a070">2</span><span style="color:#666">/</span><span style="color:#40a070">5</span>) <span style="color:#666">+</span>
  <span style="color:#06287e">xlab</span>(<span style="color:#06287e">expression</span>(<span style="color:#4070a0">&#34;log&#34;</span>[2]<span style="color:#666">*</span><span style="color:#4070a0">&#34;FC&#34;</span>)) <span style="color:#666">+</span> 
  <span style="color:#06287e">ylab</span>(<span style="color:#06287e">expression</span>(<span style="color:#4070a0">&#34;-log&#34;</span>[10]<span style="color:#666">*</span><span style="color:#4070a0">&#34;FDR&#34;</span>)) <span style="color:#666">+</span>
  <span style="color:#06287e">scale_color_viridis_d</span>() <span style="color:#666">+</span>
  <span style="color:#06287e">guides</span>(colour <span style="color:#666">=</span> <span style="color:#06287e">guide_legend</span>(override.aes <span style="color:#666">=</span> <span style="color:#06287e">list</span>(size<span style="color:#666">=</span><span style="color:#40a070">1.5</span>))) 

p3
</code></pre></div><p><img src="/post/2020-08-09-visualizing-volcano-plots-in-r_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>And we can count how many genes are up- or down-regulated according to the different significance thresholds with <code>count()</code>:</p>
<div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r">data <span style="color:#666">%&gt;%</span> 
  <span style="color:#06287e">count</span>(Expression, Significance)
</code></pre></div><div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">## # A tibble: 7 x 3
##   Expression     Significance     n
##   &lt;chr&gt;          &lt;chr&gt;        &lt;int&gt;
## 1 Down-regulated FDR 0.001     1259
## 2 Down-regulated FDR 0.01       584
## 3 Down-regulated FDR 0.05       716
## 4 Unchanged      Unchanged    16317
## 5 Up-regulated   FDR 0.001      799
## 6 Up-regulated   FDR 0.01       484
## 7 Up-regulated   FDR 0.05       786
</code></pre></div><h2 id="adding-labels-to-selected-genes">Adding labels to selected genes</h2>
<p>If we labeled all of the genes, we&rsquo;d end up with a plot with overcrowded labels that wouldn&rsquo;t be possible to read. So, we could opt for labeling only the top n genes or a specific subset of them. For example:</p>
<div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r">top <span style="color:#666">&lt;-</span> <span style="color:#40a070">10</span>
top_genes <span style="color:#666">&lt;-</span> <span style="color:#06287e">bind_rows</span>(
  data <span style="color:#666">%&gt;%</span> 
    <span style="color:#06287e">filter</span>(Expression <span style="color:#666">==</span> <span style="color:#4070a0">&#39;Up-regulated&#39;</span>) <span style="color:#666">%&gt;%</span> 
    <span style="color:#06287e">arrange</span>(FDR, <span style="color:#06287e">desc</span>(<span style="color:#06287e">abs</span>(logFC))) <span style="color:#666">%&gt;%</span> 
    <span style="color:#06287e">head</span>(top),
  data <span style="color:#666">%&gt;%</span> 
    <span style="color:#06287e">filter</span>(Expression <span style="color:#666">==</span> <span style="color:#4070a0">&#39;Down-regulated&#39;</span>) <span style="color:#666">%&gt;%</span> 
    <span style="color:#06287e">arrange</span>(FDR, <span style="color:#06287e">desc</span>(<span style="color:#06287e">abs</span>(logFC))) <span style="color:#666">%&gt;%</span> 
    <span style="color:#06287e">head</span>(top)
)
top_genes
</code></pre></div><div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">## # A tibble: 20 x 6
##    Genes   logFC   PValue      FDR Expression     Significance
##    &lt;chr&gt;   &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;          &lt;chr&gt;       
##  1 144_7   12.2  2.42e-53 1.69e-49 Up-regulated   FDR 0.001   
##  2 829_3   11.9  2.98e-51 1.25e-47 Up-regulated   FDR 0.001   
##  3 127_41   7.95 1.45e-49 3.38e-46 Up-regulated   FDR 0.001   
##  4 31_39    7.72 1.42e-47 2.97e-44 Up-regulated   FDR 0.001   
##  5 146_68  11.0  5.43e-47 1.03e-43 Up-regulated   FDR 0.001   
##  6 1078_1   7.75 6.83e-47 1.10e-43 Up-regulated   FDR 0.001   
##  7 836_2    9.11 1.19e-42 1.78e-39 Up-regulated   FDR 0.001   
##  8 5_110   12.6  2.34e-42 2.89e-39 Up-regulated   FDR 0.001   
##  9 1092_3  11.8  2.34e-41 2.73e-38 Up-regulated   FDR 0.001   
## 10 81_91    6.95 3.40e-41 3.75e-38 Up-regulated   FDR 0.001   
## 11 276_4  -10.5  1.60e-61 3.35e-57 Down-regulated FDR 0.001   
## 12 62_157  -9.32 6.98e-54 7.31e-50 Down-regulated FDR 0.001   
## 13 198_6  -10.0  3.31e-53 1.73e-49 Down-regulated FDR 0.001   
## 14 44_124  -8.91 2.98e-50 1.04e-46 Down-regulated FDR 0.001   
## 15 25_31   -9.00 6.55e-50 1.96e-46 Down-regulated FDR 0.001   
## 16 12_195  -8.94 7.51e-50 1.97e-46 Down-regulated FDR 0.001   
## 17 85_19   -7.74 6.22e-47 1.08e-43 Down-regulated FDR 0.001   
## 18 57_45   -6.50 1.36e-42 1.89e-39 Down-regulated FDR 0.001   
## 19 170_33  -6.90 1.52e-42 1.99e-39 Down-regulated FDR 0.001   
## 20 468_3   -9.32 1.16e-40 1.21e-37 Down-regulated FDR 0.001
</code></pre></div><div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r">p3 <span style="color:#666">&lt;-</span>  p3 <span style="color:#666">+</span>
  <span style="color:#06287e">geom_label_repel</span>(data <span style="color:#666">=</span> top_genes,
                   mapping <span style="color:#666">=</span> <span style="color:#06287e">aes</span>(logFC, <span style="color:#666">-</span><span style="color:#06287e">log</span>(FDR,<span style="color:#40a070">10</span>), label <span style="color:#666">=</span> Genes),
                   size <span style="color:#666">=</span> <span style="color:#40a070">2</span>)
p3
</code></pre></div><p><img src="/post/2020-08-09-visualizing-volcano-plots-in-r_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p>So, only the top 10 up- and down-regulated genes are labeled, avoiding overcrowding.</p>
<h2 id="conclusion">Conclusion</h2>
<p>In this post we applied two different color schemes on a volcano plot and labeled a few genes using tools from the tidyverse.</p>
<h2 id="session-info">Session info</h2>
<div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r">info <span style="color:#666">&lt;-</span> <span style="color:#06287e">sessionInfo</span>()
<span style="color:#06287e">toLatex</span>(info, locale <span style="color:#666">=</span> <span style="color:#007020;font-weight:bold">FALSE</span>)
</code></pre></div><div class="highlight"><pre style="background-color:#f0f0f0;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">## \begin{itemize}\raggedright
##   \item R version 4.0.2 (2020-06-22), \verb|x86_64-pc-linux-gnu|
##   \item Running under: \verb|Ubuntu 18.04.5 LTS|
##   \item Matrix products: default
##   \item BLAS:   \verb|/usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1|
##   \item LAPACK: \verb|/usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1|
##   \item Base packages: base, datasets, graphics, grDevices, methods,
##     stats, utils
##   \item Other packages: dplyr~1.0.0, forcats~0.5.0, ggplot2~3.3.2,
##     ggrepel~0.8.2, purrr~0.3.4, readr~1.3.1, stringr~1.4.0,
##     tibble~3.0.3, tidyr~1.1.0, tidyverse~1.3.0
##   \item Loaded via a namespace (and not attached): assertthat~0.2.1,
##     backports~1.1.8, blob~1.2.1, blogdown~0.20, bookdown~0.20,
##     broom~0.7.0, cellranger~1.1.0, cli~2.0.2, colorspace~1.4-1,
##     compiler~4.0.2, crayon~1.3.4, curl~4.3, DBI~1.1.0, dbplyr~1.4.4,
##     digest~0.6.25, ellipsis~0.3.1, evaluate~0.14, fansi~0.4.1,
##     farver~2.0.3, fs~1.4.2, generics~0.0.2, glue~1.4.1, grid~4.0.2,
##     gtable~0.3.0, haven~2.3.1, hms~0.5.3, htmltools~0.5.0, httr~1.4.1,
##     jsonlite~1.7.0, knitr~1.29, labeling~0.3, lifecycle~0.2.0,
##     lubridate~1.7.9, magrittr~1.5, modelr~0.1.8, munsell~0.5.0,
##     pillar~1.4.6, pkgconfig~2.0.3, R6~2.4.1, Rcpp~1.0.5, readxl~1.3.1,
##     reprex~0.3.0, rlang~0.4.7, rmarkdown~2.3, rstudioapi~0.11,
##     rvest~0.3.5, scales~1.1.1, stringi~1.4.6, tidyselect~1.1.0,
##     tools~4.0.2, utf8~1.1.4, vctrs~0.3.2, viridisLite~0.3.0,
##     withr~2.2.0, xfun~0.15, xml2~1.3.2, yaml~2.2.1
## \end{itemize}
</code></pre></div>
		</div>
		
		<div class="post-tags">
			
				
					<nav class="nav tags">
							<ul class="flat">
								
								<li><a href="/tags/data-visualization">Data visualization</a></li>
								
								<li><a href="/tags/gene-expression">Gene expression</a></li>
								
								<li><a href="/tags/r">R</a></li>
								
								<li><a href="/tags/rstats">Rstats</a></li>
								
								<li><a href="/tags/tutorial">Tutorial</a></li>
								
								<li><a href="/tags/bioinformatics">Bioinformatics</a></li>
								
							</ul>
					</nav>
				
			
			
		</div>
		</div>
	<div class="footer wrapper">
	<nav class="nav">
		<div>Â© Samuel David Gamboa-Tuz, 2019-2020 <a href="https://github.com/vividvilla/ezhil">Ezhil theme</a> | Built with <a href="https://gohugo.io">Hugo</a></div>
	</nav>
</div>



<script>feather.replace()</script>
	
</body>
</html>

