<!DOCTYPE html>
<html>
<head>
	<meta charset="utf-8" />
	<meta http-equiv="X-UA-Compatible" content="IE=edge"><title>Vizualizing Volcano Plots in R - My Test Blog</title><meta name="viewport" content="width=device-width, initial-scale=1">
	<meta property="og:title" content="Vizualizing Volcano Plots in R" />
<meta property="og:description" content="A brief tutorial about how to make volcano plots in R using the tidyverse" />
<meta property="og:type" content="article" />
<meta property="og:url" content="/post/vizualizing-volcano-plots-in-r/" />
<meta property="article:published_time" content="2020-08-09T00:00:00+00:00" />
<meta property="article:modified_time" content="2020-08-09T00:00:00+00:00" />
<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="Vizualizing Volcano Plots in R"/>
<meta name="twitter:description" content="A brief tutorial about how to make volcano plots in R using the tidyverse"/>
<link href="https://fonts.googleapis.com/css?family=Ubuntu:300,400,300italic,400italic|Raleway:200,300" rel="stylesheet">

	<link rel="stylesheet" type="text/css" media="screen" href="/css/normalize.css" />
	<link rel="stylesheet" type="text/css" media="screen" href="/css/main.css" /><link rel="stylesheet" type="text/css" href="/css/dark.css" media="(prefers-color-scheme: dark)" />

	<script src="https://cdn.jsdelivr.net/npm/feather-icons/dist/feather.min.js"></script>
	<script src="/js/main.js"></script>
</head>

<body>
	<div class="container wrapper post">
		<div class="header">
	<base href="/">
	<h1 class="site-title"><a href="/">My Test Blog</a></h1>
	<div class="site-description"><h2>Posts about science, bioinformatics, and data science.</h2><nav class="nav social">
			<ul class="flat"><a href="https://github.com/sdgamboa" title="Github"><i data-feather="github"></i></a><a href="https://twitter.com/samueldgamboa" title="Twitter"><i data-feather="twitter"></i></a><a href="https://sdgamboa.github.io/" title="My Academic Website"><i data-feather="user"></i></a><a href="/index.xml" title="RSS"><i data-feather="rss"></i></a></ul>
		</nav>
	</div>

	<nav class="nav">
		<ul class="flat">
			
			<li>
				<a href="/">Home</a>
			</li>
			
			<li>
				<a href="/post">All posts</a>
			</li>
			
			<li>
				<a href="/about">About</a>
			</li>
			
			<li>
				<a href="/tags">Tags</a>
			</li>
			
		</ul>
	</nav>
</div>


		<div class="post-header">
			<h1 class="title">Vizualizing Volcano Plots in R</h1>
			<div class="meta">Posted at &mdash; Aug 9, 2020</div>
		</div>

		<div class="markdown">
			<div class="highlight"><pre style="color:#93a1a1;background-color:#002b36;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">knitr::opts_chunk$set(echo = TRUE)
</code></pre></div><h3 id="1-introduction">1. Introduction</h3>
<p>A <strong>volcano plot</strong> is a type of scatter plot commonly used to represent expression changes of thousands of genes between two samples or treatments. Typically, this kind of plots depict the <strong>fold change</strong>, positive or negative, in the x axis vs a <strong>significance value</strong>, such as the p-value or the adjusted p-value (FDR), in the y axis. Dots represent genes, and they can be labeled or colored according to some attribute, such as whether they are up- or down-regulated, a significance threshold, or an identifier.</p>
<p>In this post I will give some examples on how to make volcano plots in <strong>R</strong> using <strong>ggplot2</strong> and some other packages from the <strong>tidyverse</strong>, such as <strong>dplyr</strong> and <strong>tidyr</strong>. I will also give an example on how to turn your R code into a <strong>bash command</strong> that can be used to make volcano plots from multiple data files at once.</p>
<h3 id="2-making-a-volcano-plot">2. Making a volcano plot</h3>
<p>To make our volcano plots we will be using the tidyverse and ggrepel packages.</p>
<div class="highlight"><pre style="color:#93a1a1;background-color:#002b36;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">library(tidyverse)
library(ggrepel)
</code></pre></div><div class="highlight"><pre style="color:#93a1a1;background-color:#002b36;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">my_data &lt;- read_tsv(&#34;/home/usuario/Projects/blog/my_ds_blog/content/post/_files/L0_vs_L20.tsv&#34;)
my_data
</code></pre></div><p>Making the plot with ggplot2</p>
<div class="highlight"><pre style="color:#93a1a1;background-color:#002b36;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">my_plot1 &lt;- ggplot(my_data, aes(logFC, -log(FDR,10))) +
  geom_point(size = 2/5) +
  xlab(expression(&#34;log&#34;[2]*&#34;FC&#34;)) + 
  ylab(expression(&#34;-log&#34;[10]*&#34;FDR&#34;)) +
  theme_bw()
my_plot1
</code></pre></div><h3 id="3-adding-color-to-up--and-down-regulated-genes">3. Adding color to up- and down-regulated genes</h3>
<p>Now only up-regulated and down-regulated</p>
<div class="highlight"><pre style="color:#93a1a1;background-color:#002b36;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">my_data &lt;- my_data %&gt;% 
  mutate(
    Expression = ifelse(
      logFC &gt;= log(2) &amp; FDR &lt;= 0.05, &#34;Up-regulated&#34;, ifelse(
        logFC &lt;= -log(2) &amp; FDR &lt;= 0.05, &#34;Down-regulated&#34;, &#34;Unchanged&#34;
      )
    )
  )
my_data
</code></pre></div><p>Now plotting</p>
<div class="highlight"><pre style="color:#93a1a1;background-color:#002b36;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">my_plot2 &lt;- ggplot(my_data, aes(logFC, -log(FDR,10))) +
  geom_point(aes(color = Expression), size = 2/5) +
  xlab(expression(&#34;log&#34;[2]*&#34;FC&#34;)) + 
  ylab(expression(&#34;-log&#34;[10]*&#34;FDR&#34;)) +
  theme_bw() +
  scale_color_manual(values = c(&#34;dodgerblue3&#34;, &#34;gray50&#34;, &#34;firebrick3&#34;)) +
  guides(colour = guide_legend(override.aes = list(size=1.5))) +
  theme(legend.title = element_blank())
my_plot2
</code></pre></div><h3 id="4-adding-labels-with-ggrepel">4. Adding labels with ggrepel</h3>
<p>We are gonna get the top up- and down-regulated genes</p>
<div class="highlight"><pre style="color:#93a1a1;background-color:#002b36;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">top &lt;- 10
top_genes &lt;- bind_rows(
  my_data %&gt;% 
    filter(Expression == &#39;Up-regulated&#39;) %&gt;% 
    arrange(FDR, desc(abs(logFC))) %&gt;% 
    head(top),
  my_data %&gt;% 
    filter(Expression == &#39;Down-regulated&#39;) %&gt;% 
    arrange(FDR, desc(abs(logFC))) %&gt;% 
    head(top)
)
top_genes
</code></pre></div><p>Now adding the labels</p>
<div class="highlight"><pre style="color:#93a1a1;background-color:#002b36;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">my_plot2 &lt;-  my_plot2 +
  geom_label_repel(data = top_genes, mapping = aes(logFC, -log(FDR,10), label = Genes),
                   size = 2)
my_plot2

</code></pre></div><h3 id="5-adding-counts-to-the-legend">5. Adding counts to the legend</h3>
<div class="highlight"><pre style="color:#93a1a1;background-color:#002b36;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">my_data_exp_counts &lt;- my_data %&gt;% 
  group_by(Expression) %&gt;% 
  summarise(Count = n())
my_data_exp_counts  
</code></pre></div><div class="highlight"><pre style="color:#93a1a1;background-color:#002b36;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">up_genes &lt;- format(my_data_exp_counts$Count[1], big.mark = &#39;,&#39;)
unchanged_genes &lt;- format(my_data_exp_counts$Count[2], big.mark = &#39;,&#39;)
down_genes &lt;- format(my_data_exp_counts$Count[3], big.mark = &#39;,&#39;)
</code></pre></div><div class="highlight"><pre style="color:#93a1a1;background-color:#002b36;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">my_plot3 &lt;- ggplot(my_data, aes(logFC, -log(FDR,10))) +
  geom_point(aes(color = Expression), size = 2/5) +
  xlab(expression(&#34;log&#34;[2]*&#34;FC&#34;)) + 
  ylab(expression(&#34;-log&#34;[10]*&#34;FDR&#34;)) +
  theme_bw() +
  scale_color_manual(labels = c(paste(&#34;Up-regulated (&#34;, up_genes, &#34;)&#34;, sep = &#34;&#34;),
                                paste(&#34;Unchanged (&#34;, unchanged_genes, &#34;)&#34;, sep = &#34;&#34;),
                                paste(&#34;Down-regulated (&#34;, down_genes, &#34;)&#34;, sep = &#34;&#34;)),
                     values = c(&#34;dodgerblue3&#34;, &#34;gray50&#34;, &#34;firebrick3&#34;) 
  ) +
  guides(colour = guide_legend(override.aes = list(size=1.5))) +
  theme(legend.title = element_blank()) +
  geom_label_repel(data = top_genes, mapping = aes(logFC, -log(FDR,10), label = Genes),
                   size = 2)
my_plot3
</code></pre></div><h3 id="6-coloring-by-significance-value">6. Coloring by significance value</h3>
<div class="highlight"><pre style="color:#93a1a1;background-color:#002b36;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">my_data &lt;- my_data %&gt;% 
  mutate(
    Significance = ifelse(
      abs(logFC) &gt;= log(2) &amp; FDR &lt;= 0.05 &amp; FDR &gt; 0.01, &#34;FDR 0.05&#34;, ifelse(
        abs(logFC) &gt;= log(2) &amp; FDR &lt;= 0.01 &amp; FDR &gt; 0.001, &#34;FDR 0.01&#34;, ifelse(
          abs(logFC) &gt;= log(2) &amp; FDR &lt;= 0.001, &#34;FDR 0.001&#34;, &#34;Unchanged&#34;
        )
      )
    )
  )
my_data
</code></pre></div><p>Let&rsquo;s known how any</p>
<div class="highlight"><pre style="color:#93a1a1;background-color:#002b36;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">my_data_exp_sig_counts &lt;- my_data %&gt;% 
  group_by(Expression, Significance) %&gt;% 
  summarise(Count = n())
my_data_exp_sig_counts  
</code></pre></div><div class="highlight"><pre style="color:#93a1a1;background-color:#002b36;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">
down001 &lt;- format(my_data_exp_sig_counts[[3]][1], big.mark = &#39;,&#39;)
down01 &lt;- format(my_data_exp_sig_counts[[3]][2], big.mark = &#39;,&#39;)
down05 &lt;- format(my_data_exp_sig_counts[[3]][3], big.mark = &#39;,&#39;)
unchanged &lt;- format(my_data_exp_sig_counts[[3]][4], big.mark = &#39;,&#39;)
up001 &lt;- format(my_data_exp_sig_counts[[3]][5], big.mark = &#39;,&#39;)
up01 &lt;- format(my_data_exp_sig_counts[[3]][6], big.mark = &#39;,&#39;)
up05 &lt;- format(my_data_exp_sig_counts[[3]][7], big.mark = &#39;,&#39;)


</code></pre></div><p>You can also embed plots, for example:</p>
<div class="highlight"><pre style="color:#93a1a1;background-color:#002b36;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-fallback" data-lang="fallback">my_plot4 &lt;- ggplot(my_data, aes(logFC, -log(FDR,10))) +
  geom_point(aes(color = Significance), size = 2/5) +
  xlab(expression(&#34;log&#34;[2]*&#34;FC&#34;)) + 
  ylab(expression(&#34;-log&#34;[10]*&#34;FDR&#34;)) +
  theme_bw() +
  scale_color_manual(labels = c(paste(&#34;FDR 0.001\n&#34;,&#34;(&#34;,down001,&#34; / &#34;,up001,&#34;)&#34;,&#34;\n&#34;, sep = &#34;&#34;),
                                paste(&#34;FDR 0.01\n&#34;,&#34;(&#34;,down01,&#34; / &#34;,up01,&#34;)&#34;,&#34;\n&#34;, sep = &#34;&#34;),
                                paste(&#34;FDR 0.05\n&#34;,&#34;(&#34;,down05,&#34; / &#34;,up05,&#34;)&#34;,&#34;\n&#34;, sep = &#34;&#34;),
                                paste(&#34;Unchanged\n&#34;,&#34;(&#34;,unchanged,&#34;)&#34;, sep = &#34;&#34;)),
                     values = c(&#34;#980043&#34;, &#34;#dd1c77&#34;, &#34;#df65b0&#34;, &#34;#d7b5d8&#34;)) +
  guides(colour = guide_legend(override.aes = list(size=1.5))) +
  theme(legend.title = element_blank()) +
  geom_label_repel(data = top_genes, mapping = aes(logFC, -log(FDR,10), label = Genes),
                   size = 2)
my_plot4
</code></pre></div>
		</div>

		<div class="post-tags">
			
				
					<nav class="nav tags">
							<ul class="flat">
								
								<li><a href="/tags/gene-expression">Gene expression</a></li>
								
								<li><a href="/tags/r">R</a></li>
								
								<li><a href="/tags/rstats">rstats</a></li>
								
								<li><a href="/tags/dataviz">DataViz</a></li>
								
							</ul>
					</nav>
				
			
		</div>
		<div id="disqus_thread"></div>
<script type="text/javascript">
	(function () {
		
		
		if (window.location.hostname == "localhost")
			return;

		var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
		var disqus_shortname = 'ezhil-demo';
		dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
		(document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
	})();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by
		Disqus.</a></noscript>
<a href="http://disqus.com/" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</div>
	<div class="footer wrapper">
	<nav class="nav">
		<div> © Samuel D. Gamboa-Tuz, 2020 |  <a href="https://github.com/vividvilla/ezhil">Ezhil theme</a> | Built with <a href="https://gohugo.io">Hugo</a></div>
	</nav>
</div>



<script>feather.replace()</script>
</body>
</html>
